packages/menu/src/SubMenu.tsx

```diff
@@ -324,32 +231,53 @@ const SubMenu: ExtendableBox<
     });
 
     return (
-      <li role="none" className={liStyle}>
-        {renderBox}
-        <IconButton
-          {...iconButton.prop}
-          darkMode={true}
-          ref={setIconButtonElement}
-          aria-label={open ? 'Close Sub-menu' : 'Open Sub-menu'}
-          className={cx(iconButtonStyle, {
-            [openIconButtonStyle]: open,
-            [iconButtonFocusedStyle]: showFocus,
-          })}
-          onClick={(e: React.MouseEvent) => {
-            // we stop the event from propagating and closing the entire menu
-            e.nativeEvent.stopImmediatePropagation();
-
-            if (setOpen) {
-              setOpen(!open);
-            }
-          }}
+      <>
+        <MenuItem
+          {...rest}
+          {...subMenuContainer.prop}
+          aria-haspopup={true}
+          onClick={onRootClick}
+          ref={ref}
+          className={cx(
+            subMenuStyle,
+            {
+              [subMenuOpenStyle]: open,
+            },
+            className,
+          )}
+          disabled={disabled}
+          active={active}
+          size={'default'}
+          description={description}
+          glyph={glyph}
         >
-          <ChevronIcon
-            role="presentation"
-            className={chevronIconStyles}
-            size={14}
-          />
-        </IconButton>
+          {title}
+          <IconButton
```

Comment:
The fact that adding this absolute-ly positioned button as a child of `MenuItem` is a bit lucky. If we add `position: relative;` to anything in the `MenuItem` in the future, this will break.

A more future-proof solution would be
1) Add a `rightGlyph` prop to `MenuItem`
2) separate the item contents into a 3rd internal component that is then consumed by both `MenuItem` and `SubMenu`

Reply:
I'm a bit confused why it's lucky. Of course it will break if we change position properties of MenuItem, but since it's dependent on other styles within the same package, we should be able to manage that change fine without exposing a new public API or making the implementation more complicated.